docs/cookbook/stockExchangeTimeZone.mjs

```diff
@@ -0,0 +1,249 @@
+  const isoDateTimePartString = (part) => `00${part}`.slice(-2);
+  const sign = offsetNanoseconds < 0 ? '-' : '+';
+  offsetNanoseconds = Math.abs(offsetNanoseconds);
+  const nanoseconds = offsetNanoseconds % 1e9;
+  const seconds = Math.floor(offsetNanoseconds / 1e9) % 60;
+  const minutes = Math.floor(offsetNanoseconds / 60e9) % 60;
+  const hours = Math.floor(offsetNanoseconds / 3600e9);
+
+  const hourString = isoDateTimePartString(hours);
+  const minuteString = isoDateTimePartString(minutes);
+  const secondString = isoDateTimePartString(seconds);
+  let post = '';
+  if (nanoseconds) {
+    let fraction = `${nanoseconds}`.padStart(9, '0');
+    while (fraction[fraction.length - 1] === '0') fraction = fraction.slice(0, -1);
+    post = `:${secondString}.${fraction}`;
+  } else if (seconds) {
+    post = `:${secondString}`;
+  }
+  return `${sign}${hourString}:${minuteString}${post}`;
+}
+
+const tzNYSE = Object.freeze({
+  getPossibleInstantsFor(dt) {
+    dt = Temporal.PlainDateTime.from(dt);
+    const zdt = dt.toZonedDateTime(tz);
+    const zdtWhenMarketIsOpen = isDuringMarketHours(zdt) ? zdt : getNextMarketOpen(zdt.toInstant());
+    return [zdtWhenMarketIsOpen.toInstant()];
+  },
+  getInstantFor(dt) {
+    dt = Temporal.PlainDateTime.from(dt);
+    // `disambiguation` option is ignored. If the market is closed, then return the
+    // opening time of the next market day.
+    const zdt = dt.toZonedDateTime(tz);
+    const zdtWhenMarketIsOpen = isDuringMarketHours(zdt) ? zdt : getNextMarketOpen(zdt.toInstant());
+    return zdtWhenMarketIsOpen.toInstant();
+  },
+  getPlainDateTimeFor(instant) {
+    instant = Temporal.Instant.from(instant);
+    const zdt = instant.toZonedDateTimeISO(tz);
+    const zdtWhenMarketIsOpen = isDuringMarketHours(zdt) ? zdt : getNextMarketOpen(zdt.toInstant());
+    return zdtWhenMarketIsOpen.toPlainDateTime();
+  },
+  getNextTransition(instant) {
+    instant = Temporal.Instant.from(instant);
+    const nextOpen = getNextMarketOpen(instant);
+    const nextClose = getNextMarketClose(instant);
+    const zdtTransition = [nextOpen, nextClose].sort(Temporal.ZonedDateTime.compare)[0];
+    return zdtTransition.toInstant();
+  },
+  getPreviousTransition(instant) {
+    instant = Temporal.Instant.from(instant);
+    const prevOpen = getPreviousMarketOpen(instant);
+    const prevClose = getPreviousMarketClose(instant);
+    const zdtTransition = [prevOpen, prevClose].sort(Temporal.ZonedDateTime.compare)[1];
+    return zdtTransition.toInstant();
+  },
+  getOffsetNanosecondsFor(instant) {
+    instant = Temporal.Instant.from(instant);
+    const zdt = instant.toZonedDateTimeISO(tz);
+    const zdtWhenMarketIsOpen = isDuringMarketHours(zdt) ? zdt : getNextMarketOpen(zdt.toInstant());
+    const ns = zdt.offsetNanoseconds + zdt.until(zdtWhenMarketIsOpen, { largestUnit: 'nanoseconds' }).nanoseconds;
+    return ns;
+  },
+  getOffsetStringFor(instant) {
```

Comment:
I'm not able to look at the whole thing just now, but it did occur to me that if you subclass Temporal.TimeZone instead of going the plain-object route, you get this for free and you don't have to write your own `formatTimeZoneOffsetString`.

Reply:
Good idea. Done!